### PR TITLE
Only format an open curly if it the first token on a line.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/4435

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

File: src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs [hunk 1]
```diff
@@ -138,18 +138,45 @@ private static bool TokenShouldNotFormatOnReturn(SyntaxToken token)
...
+            if (token.IsKind(SyntaxKind.OpenBraceToken))
+            {
+                var text = await token.SyntaxTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                if (!token.IsFirstTokenOnLine(text))
+                {
+                    return true;
+                }
+            }
+
+            return false;
```
Comment (DustinCampbell) [inline src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs]: Nicely broken apart and commented. :smile:

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Pull request merged**